deptool.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/python
+# Copyright (c) 2018 SUSE LLC
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from pprint import pprint
+import os
+import sys
+import re
+import logging
+import cmdln
+
+from fnmatch import fnmatch
+from ConfigParser import SafeConfigParser
+import solv
+import rpm
+
+logger = logging.getLogger()
+
+REASONS = [i[14:] for i in dir(solv.Solver) if i.startswith('SOLVER_REASON_')]
+
+class DepTool(cmdln.Cmdln):
+    def __init__(self, *args, **kwargs):
+        cmdln.Cmdln.__init__(self, args, kwargs)
+
+    def get_optparser(self):
+        parser = cmdln.CmdlnOptionParser(self)
+        parser.add_option("--dry", action="store_true", help="dry run")
+        parser.add_option("--debug", action="store_true", help="debug output")
+        parser.add_option("--verbose", action="store_true", help="verbose")
+        parser.add_option("--system", action="store_true", help="with system repo")
+        parser.add_option("--arch", dest="arch", help="architecture", default='x86_64')
+        return parser
+
+    def postoptparse(self):
+        level = None
+        if self.options.debug:
+            level = logging.DEBUG
+        elif self.options.verbose:
+            level = logging.INFO
+
+        logging.basicConfig(level=level)
+
+        if level:
+            logger.setLevel(level)  # wtf?
```

Comment:
Presumably this is due to calling `getLogger()` before `basicConfig()` is invoked. For example `ReviewBot.py`:

https://github.com/openSUSE/osc-plugin-factory/blob/945bf4f58ff7c66e006cb5d756e252a59bc2ede8/ReviewBot.py#L559-L560

Reply:
IIRC calling getLogger on module level is the recommended use. The way ReviewBot does it with self.logger is wrong, just me not knowing better and  not bothering to fix it after all.

Maybe the script needs to not initialize the global logger if __name__ == '__main__' so it can be used as module or program.

Reply:
That would be another way.